modules/workflow-workflowoperation/src/main/java/org/opencastproject/workflow/handler/workflow/StartWorkflowWorkflowOperationHandler.java

```diff
@@ -0,0 +1,129 @@
+import static org.apache.commons.lang3.StringUtils.trimToEmpty;
+
+import org.opencastproject.assetmanager.api.AssetManager;
+import org.opencastproject.assetmanager.api.fn.Enrichments;
+import org.opencastproject.assetmanager.api.query.AQueryBuilder;
+import org.opencastproject.assetmanager.api.query.AResult;
+import org.opencastproject.job.api.JobContext;
+import org.opencastproject.mediapackage.MediaPackage;
+import org.opencastproject.util.NotFoundException;
+import org.opencastproject.workflow.api.AbstractWorkflowOperationHandler;
+import org.opencastproject.workflow.api.WorkflowDefinition;
+import org.opencastproject.workflow.api.WorkflowInstance;
+import org.opencastproject.workflow.api.WorkflowOperationException;
+import org.opencastproject.workflow.api.WorkflowOperationInstance;
+import org.opencastproject.workflow.api.WorkflowOperationResult;
+import org.opencastproject.workflow.api.WorkflowService;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This WOH starts a new workflow for given media package.
+ */
+public class StartWorkflowWorkflowOperationHandler extends AbstractWorkflowOperationHandler {
+  private static final Logger logger = LoggerFactory.getLogger(StartWorkflowWorkflowOperationHandler.class);
+
+  /** Name of the configuration option that provides the media package ID */
+  public static final String MEDIA_PACKAGE_ID = "media-package";
+
+  /** Name of the configuration option that provides the workflow definition ID */
+  public static final String WORKFLOW_DEFINITION = "workflow-definition";
+
+  private AssetManager assetManager;
+
+  private WorkflowService workflowService;
+
+  /**
+   * Callback for the OSGi declarative services configuration.
+   *
+   * @param assetManager
+   *          the asset manager
+   */
+  public void setAssetManager(AssetManager assetManager) {
+    this.assetManager = assetManager;
+  }
+
+  /**
+   * Callback for the OSGi declarative services configuration.
+   *
+   * @param workflowService
+   *          the workflow service
+   */
+  public void setWorkflowService(WorkflowService workflowService) {
+    this.workflowService = workflowService;
+  }
+
+  @Override
+  public WorkflowOperationResult start(WorkflowInstance workflowInstance, JobContext context)
+          throws WorkflowOperationException {
+
+    final WorkflowOperationInstance operation = workflowInstance.getCurrentOperation();
+    final String configuredMediaPackageID = trimToEmpty(operation.getConfiguration(MEDIA_PACKAGE_ID));
+    final String configuredWorkflowDefinition = trimToEmpty(operation.getConfiguration(WORKFLOW_DEFINITION));
+
+    // Get media package
+    final AQueryBuilder q = assetManager.createQuery();
+    final AResult r = q.select(q.snapshot())
+                       .where(q.mediaPackageId(configuredMediaPackageID).and(q.version().isLatest()))
+                       .run();
+    if (r.getSize() == 0) {
```

Comment:
While this shouldn't happen, this could theoretically return multiple results.
Maybe use:

```java
if (r.getSize() != 1) {
```